### PR TITLE
schaeferdev layout for hhkb implementing movement layer (VIM style + Mouse)

### PR DESCRIPTION
**Pull request by schaefer-dev**
## Description
Added new HHKB layout which implements movement layer (VIM style + Mouse) while staying close to homerow. Contains some comfort-changes, particularly for Germans.

I am not entirely sure how I can put the correct Label on this PR, would be great if somebody could explain, or do it for me, thanks!

## Types of changes
* [ ]  Core
* [ ]  Bugfix
* [ ]  New Feature
* [ ]  Enhancement/Optimization
* [ ]  Keyboard (addition or update)
* [x]  Keymap/Layout/Userspace (addition or update)
* [ ]  Documentation

## Checklist:
* [x]  My code follows the code style of this project.
* [ ]  My change requires a change to the documentation.
* [ ]  I have updated the documentation accordingly.
* [x]  I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
* [ ]  I have added tests to cover my changes.
* [x]  I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hhkb/keymaps/schaeferdev/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Based off of this section:
+// https://github.com/qmk/qmk_firmware/blob/master/doc/BUILD_GUIDE.md#the-configh-file
+#ifndef CONFIG_SCHAEFERDEV_H
```
Comment (drashna) [inline keyboards/hhkb/keymaps/schaeferdev/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/hhkb/keymaps/schaeferdev/config.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#define CONFIG_SCHAEFERDEV_H
```
Comment (drashna) [inline keyboards/hhkb/keymaps/schaeferdev/config.h]:
```suggestion
```

File: keyboards/hhkb/keymaps/schaeferdev/config.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+// Bring in original defaults
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/hhkb/keymaps/schaeferdev/config.h]: This line is not actually needed, and should be removed. 

```suggestion
```

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/hhkb/keymaps/schaeferdev/config.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+#define MOUSEKEY_DELAY          0
+#define MOUSEKEY_INTERVAL       4
+#define MOUSEKEY_MAX_SPEED      1
+#define MOUSEKEY_TIME_TO_MAX    70
+#define MOUSEKEY_WHEEL_DELAY    0
+#define MOUSEKEY_WHEEL_MAX_SPEED   160
+#define MOUSEKEY_WHEEL_TIME_TO_MAX 200
+
+
+#endif
```
Comment (drashna) [inline keyboards/hhkb/keymaps/schaeferdev/config.h]:
```suggestion
```

File: keyboards/hhkb/keymaps/schaeferdev/keymap.c [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+     */
+
+    [MOVEMENT] = LAYOUT(
+        KC_GRV, KC_ACL0, KC_ACL1, KC_ACL2, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_BTN2, KC_TRNS, KC_MS_U, KC_TRNS, KC_PGUP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_BTN1, KC_MS_L, KC_MS_D, KC_MS_R, KC_PGDN, KC_LEFT, KC_DOWN, KC_UP, KC_RIGHT, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_BTN3, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TILD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
```
Comment (drashna) [inline keyboards/hhkb/keymaps/schaeferdev/keymap.c]: Could you remove the `action_get_macro` function?  It's been deprecated in favor of `process_record_user`.

Comment (schaefer-dev) [reply keyboards/hhkb/keymaps/schaeferdev/keymap.c]: Thanks for your suggestions and review!

Even after looking through documentation, I am not quite sure how replacing `action_get_macro` should work.

I also wasn't able to find any hhkb layout that doesnt implement the `action_get_macro` function. Could you lead me into the right direction?

Comment (drashna) [reply keyboards/hhkb/keymaps/schaeferdev/keymap.c]: Well, for you, it's not doing anything, actually. 

So, you could remove it without causing any issues.

As for replacing it, you'd use `process_record_user` which includes much of the same functionality, but also allows for control over built in keycodes, too. 

https://docs.qmk.fm/#/feature_macros

Comment (schaefer-dev) [reply keyboards/hhkb/keymaps/schaeferdev/keymap.c]: ah okay, I read over https://docs.qmk.fm/#/feature_macros?id=the-old-way-macro-action_get_macro and it was not quite clear to me how my `case 0:` would be represented using the 'new way of doing it'.

I will remove this function entirely, thanks again!

Comment (drashna) [reply keyboards/hhkb/keymaps/schaeferdev/keymap.c]: Ah, yeah, I can understand that.  But for the `process_record`, you're actually adding new keycodes that can be used.  Or you can modify the functionality of existing keycodes. It's very neat. :)

Comment (schaefer-dev) [reply keyboards/hhkb/keymaps/schaeferdev/keymap.c]: `action_get_macro` is now removed entirely for my layout.

I see, so basically I will only need cases that cover my defined macros, everything else will simply`return true` and I should be perfectly fine right?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**